### ISSUE TITLE
Pull stream from URL

### ISSUE THREAD

**Issue by lazzarello** [owncast/owncast#2907]
I'm building a small streaming platform called [Legal Content](https://streaming.legalcontent.live/) based on Owncast and some nginx proxies. While I could write my own authentication framework to get OBS to pass through the proxy network and push to Owncast...it would be great if Owncast could pull from a URL like the nginx_rtmp_module does. I'm already running nginx on the same server for SSL termination and Let's Encrypt certificate validation, so it's right there.

I suspect I could also do this with yet another proxy but I figured this might be a good first project to get familiar with the new Owncast codebase. I haven't upgraded to the dev branch yet and I've contributed to a couple open source projects in Go but that was years ago.

Basically the idea is the URL/key in OBS is not connected to the URL/key in Owncast, effectively creating a broadcast "switch" of sorts to move a streaming from one front-end to another.

**Comment by aaronpk** [comment]
If you're already using the nginx rtmp module, I'd recommend doing this sort of thing there instead of modifying Owncast. You can use the nginx rtmp module to create your own stream keys for different users with your own method of authentication for them too.

I'm assuming you're looking for something that lets multiple people stream to the owncast instance? In which case you can definitely create your own stream key management thing for the nginx rtmp module, where you would end up having nginx push the stream to Owncast with the main Owncast stream key.

**Comment by gabek** [comment]
I understand you have a specific use case for this, but the standard use case of Owncast wouldn't need to pull RTMP from another server. You can see the target audiences in the [product definition](https://github.com/owncast/owncast/blob/develop/docs/product-definition.md) doc if it helps clear up where I'm coming from.

It sounds like you already have solutions for such things and can accomplish this outside of Owncast, so that's certainly the best way to move forward.

**Comment by lazzarello** [comment]
awesome, thanks for your help.
